src/V3PreLex.l

```diff
@@ -112,6 +116,47 @@ bom             [\357\273\277]
 <INITIAL>"`undef"       { FL_FWDC; return VP_UNDEF; }
 <INITIAL>"`undefineall" { FL_FWDC; return VP_UNDEFINEALL; }
 <INITIAL>"`error"       { FL_FWDC; if (!pedantic()) return VP_ERROR; else return VP_DEFREF; }
+    /* We wanted this to be next to `protect But it must be before `pramga */
+    /* we win only because we both match to the end of the line so the length */
+    /* is equal and we are first*/
+<INITIAL>"`pragma"{wsn}+"protect"[^\n\r]*[\n\r] { FL_FWDC; linenoInc();
+                          if (v3Global.opt.preprocOnly()) {
+                              return VP_TEXT;
+                          }
+                          int res, yypos = 7; /* length of `pragma */
+                          char *yyt = yytext;
+                          yyt[yyleng] = 0; /* this is OK because the last char here is \n or \r */
+                          yyt += yypos;
+                          if ((res = sscanf(yyt, " protect %n", &yypos)) != 0)
+                            yyerrorf("scanf failure");
+                          yyt += yypos;
+                          if (bytes == 0) {
+                            res = sscanf(yyt, "encoding = (enctype = \"BASE64\", line_length = %d, bytes = %d)%n", &length, &bytes, &yypos);
+                            if (res == 2)
+                                return VP_TEXT; /* silently skip it */
+                            bytes = 0;
+                          } else if ((strncmp(yyt, "key_block", 9) == 0) || (strncmp(yyt, "data_block", 10) == 0)) {
+                            BEGIN(BASE64);
+                            return VP_TEXT;
+                          } else {
+                              return VP_TEXT;
+                          }
+                        }
+<BASE64>[A-Za-z0-9+/]+[=]{0,2}[\n\r] {FL_FWDC; linenoInc();
+                        if ((yyleng-1) <= length && ((yyleng & 3) == 1)) {
+                            bytes -= (yyleng-1)/4*3;
```

Comment:
bytes should decrement on only A-Za-z0-9, not e.g. newline or carriage return.

Reply:
bytes is not decremented on `\n` or `\r`, this is why (yyleng-1) is used.
`(yyleng & 3) == 1)` is a short form for `((yyleng-1) & 3) == 0)` , indicating that a complete 4-character BASE64 sequence was accepted, and no filler `=` were in this line.

Reply:
Ok, suggest add a comment please.